service/geopmdpy_test/TestSecureFiles.py

```diff
@@ -246,6 +249,26 @@ def test_creation_bad_group_owner(self):
         self.assertTrue(os.path.exists(renamed_path))
         self.check_dir_perms(sess_path)
 
+    def test_creation_nested_dirs(self):
+        """Creation of nested directories
+
+        Test secure_make_dirs() properly sets a umask to limit
+        permissions for intermediate directories that need to
+        be created as part of the path.
+        """
+        base_path = f"{self._TEMP_DIR.name}/level1"
+        sess_path = f"{base_path}/level2"
+        os.umask(0o000)
+        perm_mode = 0o711
+        secure_make_dirs(sess_path, perm_mode)
+        self.check_dir_perms(sess_path, perm_mode)
```

Comment:
```suggestion
        self.check_dir_perms(sess_path, perm_mode)
        self.check_dir_perms(base_path, perm_mode)
```
Maybe this is redundant with what you have below?

Reply:
I don't believe so. The check below ensures the base directory doesn't get any of the denied permissions set by the umask `secure_make_dirs()` is supposed to set.

Perhaps I should just mock out `os.umask()` and verify `secure_make_dirs()` calls it with the right parameter.